UnityProject/Assets/Scripts/Util/SweetExtensions.cs

```diff
@@ -61,9 +61,9 @@ public static string ExpensiveName(this GameObject go)
 		}
 
 		var player = go.Player();
-		if (player != null && !String.IsNullOrWhiteSpace(player.Name))
+		if (player != null && !String.IsNullOrWhiteSpace(player.Script.visibleName))
 		{
-			return player.Name;
+			return player.Script.visibleName;
```

Comment:
just wondering, what happens with spectator players? Do they have visibleName?

Reply:
I'm not terribly sure. Looks like, in any case, it defaults to " " if the name hasn't been set.